### PR TITLE
Properly handle Job.move() error conditions.

### PR DESCRIPTION
**Pull request by csadorf**
Fixes issue #148 .

### CODE REVIEW

**Comment by csadorf** [comment]
@rosecers This patch improves the error handling of `Job.move()` and should at least give you a more insightful error message for the issue you encountered.

I have not changed the behavior of allowing cross-device moves, simply because they may fail and then you end up with corrupted data spaces. It's much safer to copy the jobs across devices and then manually delete the source. We can possibly automate that in a future release (v1.1), but I don't want to risk adding unsafe behavior one day before our final release.

**Comment by csadorf** [comment]
@vyasr I think the coverage decreased, because I don't test the cross-device error state, but I'm not sure how to mock that as part of the unit tests.

**Comment by vyasr** [comment]
I agree, I think the drop is OK. The raw `else..raise` at the end is also not meaningfully testable, if you knew what other error codes to check then you would add special handling behavior.

File: signac/contrib/job.py [hunk 1]
```diff
@@ -441,16 +441,33 @@ def move(self, project):
...
+        :param project:
+            The project to move this job to.
+        :type project:
+            :py:class:`~.project.Project`
+        :raises DestinationExistsError:
+            If the job is already initialized in project.
+        :raises RuntimeError:
+            If the job is not initialized or the destination is on a different
+            device.
+        :raies OSError:
```
Comment (vyasr) [inline signac/contrib/job.py]: raies->raises

Comment (csadorf) [reply signac/contrib/job.py]: fixed

**Pull request merged**